test/e2e/healthcheck_test.go

```diff
@@ -35,6 +35,9 @@ var _ = Describe("Health Check Test", Serial, Labels{"health_checks"}, func() {
 
 	var dnsRecord *v1alpha1.DNSRecord
 
+	recordsReadyMaxDuration := GetRecordsReadyTimeout(testDNSProvider)
+	recordsRemovedMaxDuration := GetRecordsDeletedTimeout(testDNSProvider)
```

Comment:
You might need to actually call these in a `BeforeX` call to be sure `testDNSProvider` is set(It could be nil here and defaulting to your lower timeouts for Azure). Since these are global, you could just set these variables once in `BeforeSuite` after the point where we know `testDNSProvider` has been [set](https://github.com/Kuadrant/dns-operator/blob/main/test/e2e/suite_test.go#L101).